docs/user-guide/deploy/deploy_to_docker/python.md

```diff
@@ -0,0 +1,318 @@
+```bash
+docker run -p 8080:8080 \
+  -e OPENAI_API_KEY=$OPENAI_API_KEY \
+  my-agent-image:latest
+```
+
+This example uses OpenAI credentials by default, but any model provider credentials can be passed as environment variables when running the image. For instance, to pass AWS credentials:
+```bash
+docker run -p 8080:8080 \
+  -e AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID \
+  -e AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY \
+  -e AWS_REGION=us-east-1 \
+  my-agent-image:latest
+```
+
+### Step 3: Test Your Deployment
+
+Test the endpoints:
+```bash
+# Health check
+curl http://localhost:8080/ping
+
+# Test agent invocation
+curl -X POST http://localhost:8080/invocations \
+  -H "Content-Type: application/json" \
+  -d '{"input": {"prompt": "What is artificial intelligence?"}}'
+```
+
+### Step 4: Making Changes
+
+When you modify your code, rebuild and run:
+
+```bash
+# Rebuild image
+docker build -t my-agent-image:latest .
+
+# Stop existing container (if running)
+docker stop $(docker ps -q --filter ancestor=my-agent-image:latest)
+
+# Run new container
+docker run -p 8080:8080 \
+  -e OPENAI_API_KEY=$OPENAI_API_KEY \
+  my-agent-image:latest
+```
+
+## Troubleshooting
+
+- **Container not starting**: Check logs with `docker logs $(docker ps -q --filter ancestor=my-agent-image:latest)`
+- **Connection refused**: Verify app is listening on 0.0.0.0:8080
+- **Image build fails**: Check `pyproject.toml` and dependencies
+- **Port already in use**: Use different port mapping `-p 8081:8080`
+
+
+## Docker Compose for Local Development
+
+**Optional**: Docker Compose is only recommended for local development. Most cloud service providers only support raw Docker commands, not Docker Compose.
+
+For local development and testing, Docker Compose provides a more convenient way to manage your container:
+
+```yaml
+# Example for OpenAI
+version: '3.8'
+services:
+  my-python-agent:
+    build: .
+    ports:
+      - "8080:8080"
+    environment:
+      - OPENAI_API_KEY=<your-api-key>
+```
+
+Run with Docker Compose:
+```bash
+# Start services
+docker-compose up --build
+
+# Run in background
+docker-compose up -d --build
+
+# Stop services
+docker-compose down
+```
+
+## Optional: Deploy to Cloud Container Service
+
+Once your application works locally with Docker, you can deploy it to any cloud-hosted container service. The Docker container you've created is the foundation for deploying to the cloud platform of your choice (AWS, GCP, Azure, etc).
+
+Our other deployment guides build on this Docker foundation to show you how to deploy to specific cloud services:
+
+- [Amazon Bedrock AgentCore](../deploy_to_bedrock_agentcore/python.md) - Deploy to AWS with Bedrock integration
+- [AWS Fargate](../deploy_to_aws_fargate.md) - Deploy to AWS's managed container service
+- [Amazon EKS](../deploy_to_amazon_eks.md) - Deploy to Kubernetes on AWS
+- [Amazon EC2](../deploy_to_amazon_ec2.md) - Deploy directly to EC2 instances
```

Comment:
In most other docs, we have Related Resources at the end - I think that makes sense here too?